tests/unit/tf/models/test_retrieval.py

```diff
@@ -54,6 +55,16 @@ def test_two_tower_model(music_streaming_data: Dataset, run_eagerly, num_epochs=
     assert len(losses.epoch) == num_epochs
     assert all(measure >= 0 for metric in losses.history for measure in losses.history[metric])
 
+    query_features = testing_utils.get_model_inputs(
+        music_streaming_data.schema.select_by_tag(Tags.USER), ["user_genres"]
+    )
+    testing_utils.test_model_signature(model.first.query_block(), query_features, ["output_1"])
```

Comment:
Why is the output for retrieval tasks `output_1`

Reply:
Good question, I actually just realized this today as well. I think if you don’t explicitly label the output, TF will call it `output_1`. I need to digg a bit deeper in order to figure out how to add custom naming.